### PR TITLE
[#24789] Make Prism the default Go SDK runner.

### PR DESCRIPTION
**Pull request by lostluck**
Final PR that sets the prism runner as the default runner for the Go SDK.

Large PR with all changes, and passing tests: see 27550. This PR should pass everything once all parts of that PR have been submitted.

See https://github.com/apache/beam/issues/24789

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by lostluck** [comment]
I suspect the Playground examples are unhappy due to the new Go 1.20 requirement that didn't hard manifest until the example changed. I'll have a separate PR for that. There's a few docker containers that assume Go 1.18...

**Comment by lostluck** [comment]
I'm on the side of "the playground shouldn't interrupt SDK development". It's trying to have things both ways "using a released version" and "using the head version", likely causing the mismatches.

File: sdks/go/test/integration/integration.go [hunk 1]
```diff
@@ -136,6 +136,39 @@ var portableFilters = []string{
 	"TestSetStateClear",
 }
 
+// TODO(lostluck): set up a specific run for these.
+var prismFilters = []string{
+	// The portable runner does not support the TestStream primitive
```
Comment (jrmccluskey) [inline sdks/go/test/integration/integration.go:141]:
```suggestion
	// The prism runner does not support the TestStream primitive
```

File: sdks/go/test/integration/integration.go [hunk 2]
```diff
@@ -136,6 +136,39 @@ var portableFilters = []string{
...
+	"TestTestStream.*",
+	// The trigger and pane tests uses TestStream
+	"TestTrigger.*",
+	"TestPanes",
+	// TODO(https://github.com/apache/beam/issues/21058): Python portable runner times out on Kafka reads.
```
Comment (jrmccluskey) [inline sdks/go/test/integration/integration.go:146]: If this issue is present for prism you'll want a more specific error message

File: sdks/go/test/integration/integration.go [hunk 3]
```diff
@@ -136,6 +136,39 @@ var portableFilters = []string{
...
+	"TestKafkaIO.*",
+	// TODO(BEAM-13215): GCP IOs currently do not work in non-Dataflow portable runners.
+	"TestBigQueryIO.*",
+	"TestSpannerIO.*",
+	// The portable runner does not support self-checkpointing
```
Comment (jrmccluskey) [inline sdks/go/test/integration/integration.go:151]:
```suggestion
	// The prism runner does not support self-checkpointing
```

File: sdks/go/test/integration/integration.go [hunk 4]
```diff
@@ -136,6 +136,39 @@ var portableFilters = []string{
...
+	"TestCheckpointing",
+	// The portable runner does not support pipeline drain for SDF.
```
Comment (jrmccluskey) [inline sdks/go/test/integration/integration.go:153]:
```suggestion
	// The prism runner does not support pipeline drain for SDF.
```

File: sdks/go/test/integration/integration.go [hunk 5]
```diff
@@ -136,6 +136,39 @@ var portableFilters = []string{
...
+	"TestDrain",
+	// FhirIO currently only supports Dataflow runner
+	"TestFhirIO.*",
+	// OOMs currently only lead to heap dumps on Dataflow runner
+	"TestOomParDo",
+	// The portable runner does not support user state.
```
Comment (jrmccluskey) [inline sdks/go/test/integration/integration.go:159]:
```suggestion
	// The prism runner does not support user state.
```

**Comment by lostluck** [comment]
I'm digging around and it's unclear to me how I'd actually run any of these things locally... there's a lot of indirection, and I'm a little afraid I'd accidently deploy the playground or something.

**Comment by lostluck** [comment]
I've overcome my fear, and got the non-cloudbuild version of this task working. Working on it.

**Comment by lostluck** [comment]
I'm confident that fixes the Go SDK issue. I have no idea about what would cause the commit step to fail though.

**Review by jrmccluskey — APPROVED** [review]
LGTM, I'll leave the merging up to you when you're confident

**Comment by lostluck** [comment]
I'm sufficiently confident that the issue is likely in how I push branches for merging in this case. Not finding the commit, but passing the previously failing container build is good enough.

**Pull request merged**